File: schemaorggo/lose_action.go
package schemaorggo

import "encoding/json"

// LoseAction see : https://schema.org/LoseAction
type LoseAction struct {
	AchieveAction

	typeContext

	// Winner see : https://schema.org/winner
	// A sub property of participant. The winner of the action.
	// types : Person
	Winner []*Person `json:"winner,omitempty"`
}

func (v LoseAction) intoMap(intop *map[string]interface{}) error {
	if intop == nil {
		return nil
	}

	v.AchieveAction.intoMap(intop)

	into := *intop

	{
		var value interface{} = v.Winner
		if len(v.Winner) == 1 {
			value = v.Winner[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["winner"] = json.RawMessage(b)
		}
	}

	*intop = into

	return nil
}

func (v LoseAction) asMap() (map[string]interface{}, error) {
	data := map[string]interface{}{}
	err := v.intoMap(&data)
	if err != nil {
		return nil, err
	}

	data["@context"] = "http://schema.org"
	data["@type"] = "LoseAction"

	return data, nil
}

func (v LoseAction) MarshalJSON() ([]byte, error) {
	data, err := v.asMap()
	if err != nil {
		return nil, err
	}

	return json.Marshal(data)
}

File: schemaorgphp/classes/lose_action.php
<?php

class LoseAction extends AchieveAction implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'LoseAction';
	
	/**
	 * A sub property of participant. The winner of the action.
	 * see : https://schema.org/winner
	 * @var \Person|\Person[]
	 */
	public var $winner;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'LoseAction'
		);
		
		$serialized = so_json_serialize( $this->winner );
		if ( ! empty( $serialized ) ) {
			$out['winner'] = $serialized;
		}
		
		return $out;
	}
}
